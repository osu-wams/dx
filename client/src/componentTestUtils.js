import React from 'react';
import { render as rtlRender } from 'react-testing-library';
import { createHistory, createMemorySource, LocationProvider } from '@reach/router';
import { ThemeProvider } from 'styled-components';
import theme from './theme';

function render(ui, ...rest) {
  return rtlRender(<ThemeProvider theme={theme}>{ui}</ThemeProvider>, ...rest);
}

// this is a handy function that I would utilize for any component
// that relies on the router being in context
function renderWithRouter(
  ui,
  { route = '/', history = createHistory(createMemorySource(route)) } = {}
) {
  return {
    ...rtlRender(<LocationProvider history={history}>{ui}</LocationProvider>),
    // adding `history` to the returned utilities to allow us to reference it in our tests
    history
  };
}

export * from 'react-testing-library';
// override the built-in render with our own
export { render, renderWithRouter };
