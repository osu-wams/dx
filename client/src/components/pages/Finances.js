import React, { Component } from 'react';
import PageTitle from '../layout/PageTitle';

export default class Finances extends Component {
  constructor(props) {
    super(props);
    this.state = { api: 'Api' };
  }

  render() {
    const { api } = this.state;
    return (
      <div data-testid="finances-page">
        <PageTitle title="Financial Information" />
        <p>Finances {api} </p>
      </div>
    );
  }
}
