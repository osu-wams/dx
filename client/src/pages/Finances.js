import React from 'react';
import PageTitle from '../ui/PageTitle';
import {
  Overview,
  OverviewContent,
  OverviewSection,
  OverviewItem,
  OverviewItemLabel
} from '../ui/Overview';
import { Table, TableBody, TableCell, TableHeader, TableRow, TableHeaderCell } from '../ui/Table';
import { formatDollars, formatDate } from '../util/helpers';

//Temporary mock data
import accountBalance from '../api/__mocks__/accountBalance.data';
import accountTransactions from '../api/__mocks__/accountTransactions.data';
import Button from '../ui/Button';

const AcademicOverview = () => {
  const balance = accountBalance[0].attributes.currentBalance;
  const transactions = accountTransactions[0].attributes.transactions;
  return (
    <Overview>
      <OverviewContent>
        <OverviewSection>
          <OverviewItem>
            <OverviewItemLabel color="pineStand">Account Balance</OverviewItemLabel>
            <span className="l">{formatDollars(balance)}</span>
          </OverviewItem>
        </OverviewSection>
        <OverviewSection>
          <OverviewItem>
            <OverviewItemLabel color="pineStand">Last 3 account transactions</OverviewItemLabel>
            <Table variant="basic" striped>
              <TableHeader>
                <TableRow>
                  <TableHeaderCell>Amount</TableHeaderCell>
                  <TableHeaderCell>Date</TableHeaderCell>
                </TableRow>
              </TableHeader>
              <TableBody>
                {transactions.slice(0, 3).map(({ amount, entryDate }) => (
                  <TableRow key={amount + entryDate}>
                    <TableCell>{formatDollars(amount)}</TableCell>
                    <TableCell>{formatDate(entryDate, 'compact')}</TableCell>
                  </TableRow>
                ))}
              </TableBody>
            </Table>
            {/* @TODO: Create another component with full the full list of details */}
            <Button bg="pineStand" size="small" outline>
              Detailed transactions
            </Button>
          </OverviewItem>
        </OverviewSection>
      </OverviewContent>
    </Overview>
  );
};

const Finances = () => {
  return (
    <div data-testid="finances-page">
      <PageTitle title="Financial Information" />
      <AcademicOverview />
    </div>
  );
};

export default Finances;
