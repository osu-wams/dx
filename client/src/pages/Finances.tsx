import React from 'react';
import PageTitle from '../ui/PageTitle';
import AnnouncementContainer from '../ui/AnnouncementContainer';

const Finances = () => {
  return (
    <div data-testid="finances-page">
      <PageTitle title="Financial Information" />
      <AnnouncementContainer type="financial" />
    </div>
  );
};

export default Finances;
