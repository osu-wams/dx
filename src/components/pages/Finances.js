import React, { Component } from 'react';

export default class Finances extends Component {
  constructor(props) {
    super(props);
    this.state = { api: 'Api' };
  }

  render() {
    const { api } = this.state;
    return <div>Finances {api} </div>;
  }
}
