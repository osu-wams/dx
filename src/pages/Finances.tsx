import React from 'react';
import { faDollarSign } from '@fortawesome/pro-light-svg-icons';
import PageTitle from '../ui/PageTitle';
import ResourcesCard from '../features/ResourcesCard';
import AnnouncementContainer from '../ui/AnnouncementContainer';
import FinancialTransactions from '../features/FinancialTransactions';
import FinancialOverview from '../features/financial-overview/FinancialOverview';
import { MainGridWrapper, Masonry } from '../theme';

const Finances = () => {
  return (
    <>
      <MainGridWrapper data-testid="finances-page">
        <PageTitle title="Finances" />
        <Masonry>
          <FinancialOverview />
          <ResourcesCard categ="financial" icon={faDollarSign} />
          <FinancialTransactions />
        </Masonry>
      </MainGridWrapper>
      <AnnouncementContainer data-testid="finances-announcements" page="finances" />
    </>
  );
};

export default Finances;
