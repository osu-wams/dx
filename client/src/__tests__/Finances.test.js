import React from 'react';
import { render } from '../componentTestUtils';
import Finances from '../components/pages/Finances';

test('renders', () => {
  const { container } = render(<Finances />);
  expect(container.firstChild).toMatchSnapshot();
});
