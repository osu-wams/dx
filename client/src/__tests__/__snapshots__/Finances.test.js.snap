// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  data-testid="finances-page"
>
  <div
    style="border: 0px; height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; position: absolute;"
  >
    <h2>
      Financial Information
    </h2>
  </div>
  <p>
    Finances 
    Api
     
  </p>
</div>
`;
